src/mscorlib/shared/System/StringSpanHelpers.cs

```diff
@@ -149,8 +151,38 @@ public static void CheckStringComparison(StringComparison comparisonType)
             // Single comparison to check if comparisonType is within [CurrentCulture .. OrdinalIgnoreCase]
             if ((uint)(comparisonType - StringComparison.CurrentCulture) > (StringComparison.OrdinalIgnoreCase - StringComparison.CurrentCulture))
             {
-                throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
+                ThrowStringComparison_NotSupported();
             }
         }
+
+        public static CompareOptions GetCaseCompareOfComparisonCulture(StringComparison comparisonType)
+        {
+            Debug.Assert((uint)(comparisonType - StringComparison.CurrentCulture) <= (StringComparison.InvariantCultureIgnoreCase - StringComparison.CurrentCulture));
+
+            // Culture enums can be & with CompareOptions.IgnoreCase 0x01 to extract if IgnoreCase or CompareOptions.None 0x00
+            //
+            // CompareOptions.None                          0x00
+            // CompareOptions.IgnoreCase                    0x01
+            //
+            // StringComparison.CurrentCulture:             0x00
+            // StringComparison.InvariantCulture:           0x02
+            //
+            // StringComparison.CurrentCultureIgnoreCase:   0x01
+            // StringComparison.InvariantCultureIgnoreCase: 0x03
+
+            return (CompareOptions)((int)comparisonType & (int)CompareOptions.IgnoreCase);
+        }
+
+        [StackTraceHidden]
```

Comment:
I am seeing this attribute for the first time. I can infer what it does, but can you explain why we should have it here?

Is it mainly to help debugging by avoiding an extra method show up in the call stack?

Reply:
It doesn't output the method in the stacktrace; so it looks like it was thrown in the original method that calls it